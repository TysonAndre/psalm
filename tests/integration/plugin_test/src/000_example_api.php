<?php

namespace {

    class BaseClass
    {
        // NOTE: For anything which is a subclass of BaseClass,
        // if there is an entry for a method name in METHOD_FILTERS (case-insensitive),
        // then the union type of the first parameter of that method will depend on the corresponding entry in METHOD_FILTERS
        const METHOD_FILTERS = [];
    }
    class ExampleAPI extends BaseClass
    {
    }
}

namespace ExampleNamespace {

    class MyClass extends \ExampleAPI
    {
        const METHOD_FILTERS = [
        'myAPIMethod' => [
            // E.g. Union Types
            'field' => 'string',
            'otherField' => 'string[]',
        ],
        ];

        /**
         * @param array $params (Will be overridden by plugin checking METHOD_FILTERS
         * @return void
         */
        public function myAPIMethod(array $params)
        {
            printf("%s\n", $params['field']);
            printf("%s\n", $params['fieldTypo']);
            echo count($params['otherField']);
            echo strlen($params['otherField']);
        }
    }
}
