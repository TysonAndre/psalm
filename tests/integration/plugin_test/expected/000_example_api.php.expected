src/000_example_api.php:30: [E0001] InvalidArrayOffset: Cannot access value on variable $params using string offset, expecting 'field' or 'otherField' (column 24)
src/000_example_api.php:32: [E0001] InvalidArgument: Argument 1 of strlen expects string, array<mixed, string> provided (column 21)
